src/include/migraphx/run_loop.hpp

```diff
@@ -0,0 +1,120 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_RUN_LOOP_HPP
+#define MIGRAPHX_GUARD_RTGLIB_RUN_LOOP_HPP
+
+#include <string>
+#include <migraphx/instruction_ref.hpp>
+#include <migraphx/shape.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/context.hpp>
+#include <migraphx/module.hpp>
+#include <migraphx/config.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+static inline std::pair<int, bool> get_name_index(const std::string& name,
+                                                  const std::string& param_prefix)
+{
+    auto loc = name.find(param_prefix);
+    if(loc != std::string::npos)
+    {
+        int index = std::stoi(name.substr(loc + param_prefix.size()));
+        return {index, true};
+    }
+
+    std::string out_prefix = "#output_";
+    loc                    = name.find(out_prefix);
+    if(loc != std::string::npos)
+    {
+        int index = std::stoi(name.substr(loc + out_prefix.size()));
+        return {index, false};
+    }
+
+    return {-1, false};
+}
+
+template <class LoopModel, class T>
+argument run_loop(const LoopModel& model,
+                  T& ctx,
+                  std::vector<argument> args,
+                  const std::vector<module_ref>& mods,
+                  const std::function<std::vector<argument>(
+                      module_ref&, const std::unordered_map<std::string, argument>&)>& run)
+{
+    std::vector<std::vector<argument>> results;
+    // process argu lists
+    auto iter_num = args.at(0).at<int64_t>();
+    auto cond     = args.at(2).at<bool>();
```

Comment:
Why dont we put the iter_num and cond as the first two parameters? This will make it easier to slice the args.